pages/claim/[type]/[code].tsx

```diff
@@ -106,7 +95,16 @@ function ClaimDonation(): ReactElement {
         });
       }
     }
-  }, [contextLoaded, user, code]);
+  }, [contextLoaded, user]);
+
+  React.useEffect(() => {
+    //redeem code using route
+    if (user && contextLoaded) {
+      if (ready && router.query.type && router.query.code) {
+        redeemingCode(router.query.code);
+      }
+    }
+  }, [user, contextLoaded, ready, router.query.type, router.query.code]);
 
   return ready ? (
     <LandingSection>
```

Comment:
For split seconds you can see redeem modal flashing on screen when redirect to login is triggered.
Along with `redeemedCodeData`, a `user` check needs to be added in order to avoid that.